storage/innobase/handler/ha_innodb.cc

```diff
@@ -13434,6 +13451,78 @@ ha_innobase::rename_table(
 	DBUG_RETURN(convert_error_code_to_mysql(error, 0, NULL));
 }
 
+/** Initialize committed count within dict_table_t.
+@return  0 or error code */
+int
+ha_innobase::enable_persistent_count()
+{
+	dict_table_t* ib_table = m_prebuilt->table;
+	uchar* read_buf = m_prebuilt->m_mysql_table->record[0];
+	trx_t* trx = m_prebuilt->trx;
+	int err;
+
+	dict_index_t* index = UT_LIST_GET_FIRST(ib_table->indexes);
+	rw_lock_sx_lock(&index->lock);
+
+	if (ib_table->committed_count_inited) {
+		rw_lock_sx_unlock(&index->lock);
+		return -1;  /* Already initialized */
+	}
+
+	ib_table->committed_count = 0;
+	rnd_init(true);
+	do {
+		err = rnd_next(read_buf);
```

Comment:
can this operation be aborted by a user? (KILL PROCESS)?

Reply:
I've changed this block to aggregate the count into a local variable `uint64_t count` before setting `ib_table->committed_count = count` at the end. Does that address the abort-ability of this block?

Reply:
I'm not sure what is required. it was an open question.